samples: mesh/onoff_level_lighting_vnd_app: improved state Binding & upgrade Vnd Model archit. & bug fixing

vikrant8052 opened the pull request:
Removed redundancy from state binding & make it simplified.
Avoid partial state binding in case of state_binding() function
get called with invalid or IGNORE arguments.

This patch also improved architecture of already implemented Vendor
Model.

Plus it will change buffer length for all GET handlers as per pattern of
 < 2 + parameters length (in bytes) + 4 >

Signed-off-by: Vikrant More <vikrant8051@gmail.com>

vikrant8051 commented:
Kindly do not merged it. I'm trying to improve vendor model capabilities.

vikrant8052 committed:
samples: mesh/onoff_level_lighting_vnd_app: Vendor Model upgrade

This patched improved architecture of already implemented Vendor
Model. Now there are 4 separate opcodes for Vendor viz;
get, set, set_unack & status. This helps to setup
<command + response> mechanism.

Signed-off-by: Vikrant More <vikrant8051@gmail.com>

vikrant8052 committed:
samples: mesh/onoff_level_lighting_vnd_app: fix buffers length

Changes buffer length for all GET handlers as per patter of
 < 2 + parameters length (in bytes) + 4 >

Signed-off-by: Vikrant More <vikrant8051@gmail.com>

vikrant8052 committed:
samples: mesh/onoff_level_lighting_vnd_app: corrected printk message

Corrected printk messgae for GET handler for Generic Power OnOff
server.

Signed-off-by: Vikrant More <vikrant8051@gmail.com>

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/mesh/device_composition.c:
```diff
@@ -56,7 +56,7 @@ BT_MESH_MODEL_PUB_DEFINE(light_lightness_cli_pub, NULL, 2 + 5);
 BT_MESH_MODEL_PUB_DEFINE(light_ctl_cli_pub, NULL, 2 + 9);
 
-BT_MESH_MODEL_PUB_DEFINE(vnd_pub, NULL, 3 + 2);
+BT_MESH_MODEL_PUB_DEFINE(vnd_pub, NULL, 3 + 6);
```
Unrelated to this PR, but why do you have a vendor model in this app? I think it may confuse people who are only looking for a reference for standard models.

vikrant8051 replied:
It is related. I'm going to rename main heading.

Developers are not going to depends on only Standard Models. My goal is to explore all
capabilities under one demo.

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/mesh/device_composition.c:
```diff
@@ -207,6 +208,13 @@ static void state_binding(u8_t lightness, u8_t temperature)
+		gen_onoff_srv_root_user_data.onoff = 1;
+	}
+
+jump:
```
Could you rename this to something more descriptive. Would `update_temp` be accurate?

vikrant8051 replied:
Yes, I did it using "git rebase -i" commands :)

vikrant8052 reviewed (commented):
For that, if I make new commit is that fine for you ? Then I will surely do it.

vikrant8052 committed:
samples: mesh/onoff_level_lighting_vnd_app: improved state binding

Removed redundancy from state binding & make it simplified.
Avoid partial state binding in case of state_binding() function
get called with invalid or IGNORE arguments.

Signed-off-by: Vikrant More <vikrant8051@gmail.com>

vikrant8051 commented:
@jhedberg All checks have passed & I've already make suggested changes. Please approved it.

Thank You !!

jhedberg commented:
@MariuszSkamra could you rerun the PTS tests with this PR (now merged)? @vikrant8051 FYI there were many test failures when @MariuszSkamra ran the tests. He will files issues soon.
